src/category_theory/sites/sheaf.lean

```diff
@@ -54,13 +54,97 @@ variables (J : grothendieck_topology C)
 -- We follow https://stacks.math.columbia.edu/tag/00VL definition 00VR
 
 /--
-A sheaf of A is a presheaf P : C^op => A such that for every X : A, the
-presheaf of types given by sending U : C to Hom_{A}(X, P U) is a sheaf of types.
+A sheaf of A is a presheaf P : Cᵒᵖ => A such that for every E : A, the
+presheaf of types given by sending U : C to Hom_{A}(E, P U) is a sheaf of types.
 
 https://stacks.math.columbia.edu/tag/00VR
 -/
 def is_sheaf (P : Cᵒᵖ ⥤ A) : Prop :=
-∀ X : A, presieve.is_sheaf J (P ⋙ coyoneda.obj (op X))
+∀ E : A, presieve.is_sheaf J (P ⋙ coyoneda.obj (op E))
+
+section limit_sheaf_condition
+open presieve presieve.family_of_elements limits
+variables (P : Cᵒᵖ ⥤ A) {X : C} (S : sieve X) (R : presieve X) (E : Aᵒᵖ)
+
+@[simps] def cones_equiv_sieve_compatible_family :
+  (functor.cones (S.arrows.diagram.op ⋙ P)).obj E ≃
+  {x : family_of_elements (P ⋙ coyoneda.obj E) S // x.sieve_compatible} :=
+{ to_fun := λ π, ⟨λ Y f h, π.app (op ⟨over.mk f, h⟩), λ _, by
+    { intros, apply (id_comp _).symm.trans, dsimp,
+      convert π.naturality (quiver.hom.op _),
+      refl, refl, refl, swap, apply over.hom_mk, refl }⟩,
+  inv_fun := λ x, { app := λ f, x.1 f.unop.1.hom f.unop.2,
+    naturality' := λ f f' g, by
+    { refine eq.trans _ (x.2 f.unop.1.hom g.unop.left f.unop.2),
+      erw id_comp, congr, rw over.w g.unop } },
+  left_inv := λ π, by { ext, dsimp, congr,
+    rw op_eq_iff_eq_unop, ext, symmetry, apply costructured_arrow.eq_mk },
+  right_inv := λ x, by { ext, refl } }
+
+variables {P S E} {x : family_of_elements (P ⋙ coyoneda.obj E) S} (hx : x.sieve_compatible)
+
+@[simp] def _root_.category_theory.presieve.family_of_elements.sieve_compatible.cone :
+  cone (S.arrows.diagram.op ⋙ P) :=
+{ X := E.unop, π := (cones_equiv_sieve_compatible_family P S E).inv_fun ⟨x,hx⟩ }
+
+def hom_equiv_amalgamation :
+  cone_morphism hx.cone (P.map_cone S.arrows.cocone.op) ≃ {t // x.is_amalgamation t} :=
```

Comment:
You should be able to use `\hom` notation instead of writing out `cone_morphism`.

Reply:
Indeed, this only served as a hint for the expected type when I haven't defined `sieve_compatible.cone`.